web/fido_app/models.py

```diff
@@ -76,20 +76,30 @@ def check_password(self, password):
     def has_password(self):
         return self.password_hash is not None
 
-    def add_session(self, session: Session, commit=False):
+    def add_session(self, flask_session, commit=False):
         '''Associates the session with the given id to this particular user'''
+        session = Session.query.filter_by(token=flask_session["token"]).first()
+        
+        if session and session.user_id:
+            # Don't add a session token that is already associated to a user
+            if session.user_id != self.id:
+                logger.warn(f"This session token is attached to {session.user}, but is also being used by {self}!")
```

Comment:
If I log into one account, log out, then log into a second account, the code enters this conditional block and no database changes occur. I just wanted to make sure: is this the desired behavior?

Reply:
Yeah, that's a fair question. Instead of doing nothing when there is a conflict, I think it will be better to dynamically change the session token if there's a conflict

Reply:
To fix this, I just dynamically create a new session token if there is a token conflict. Unfortunately, my fix can potentially misattribute one batch of interactions to the old user instead of the new user.

Either we can deal with this issue, or we can clear the session token every time a user logs out. I think that may be an easier fix.